test/integration/src/test/java/io/pravega/test/integration/SingleThreadEndToEndTest.java

```diff
@@ -84,4 +117,65 @@ public void testSealedStream() throws Exception {
         writer.flush();
     }
 
+    @Test(timeout = 10000)
+    public void testFetchEvent() throws ReinitializationRequiredException {
+        String endpoint = "localhost";
+        String streamName = "testEventPointer";
+        String readerName = "reader";
+        String readerGroup = "testEventPointer-group";
+        int port = TestUtils.getAvailableListenPort();
+        String testString = "fetchEvent-teststring ";
+        String scope = "Scope1";
+        StreamSegmentStore store = SERVICE_BUILDER.createStreamSegmentService();
+        TableStore tableStore = SERVICE_BUILDER.createTableStoreService();
+        @Cleanup
+        PravegaConnectionListener server = new PravegaConnectionListener(false, port, store, tableStore, SERVICE_BUILDER.getLowPriorityExecutor());
+        server.startListening();
+
+        @Cleanup
+        SocketConnectionFactoryImpl clientCF = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
+        @Cleanup
+        ConnectionPoolImpl pool = new ConnectionPoolImpl(ClientConfig.builder().build(), clientCF);
+        MockController controller = new MockController(endpoint, port, pool, true);
+
+        @Cleanup
+        StreamManager streamManager = new StreamManagerImpl(controller, pool);
+        @Cleanup
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(scope, controller, ClientConfig.builder().build());
+        ReaderGroupManager readerGroupManager = new ReaderGroupManagerImpl(scope, controller, clientFactory);
+
+        streamManager.createScope(scope);
+        streamManager.createStream(scope, streamName, StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.fixed(1))
+                .build());
+        readerGroupManager.createReaderGroup(readerGroup, ReaderGroupConfig
+                .builder()
+                .stream(Stream.of(scope, streamName))
+                .automaticCheckpointIntervalMillis(2000)
+                .build());
+        UTF8StringSerializer serializer = new UTF8StringSerializer();
+
+        @Cleanup
+        EventStreamWriter<String> producer = clientFactory.createEventWriter(streamName, serializer, EventWriterConfig.builder().build());
+
+        int writeCount = 0;
+        for ( int eventNumber = 1; eventNumber <= 10; eventNumber++ ) {
+            producer.writeEvent(testString);
+            writeCount++;
+        }
+        producer.flush();
+        @Cleanup
+        EventStreamReader<String> reader = clientFactory.createReader(readerName, readerGroup, serializer, ReaderConfig.builder().build());
+        EventPointer pointer = reader.readNextEvent(100).getEventPointer();
+        AtomicInteger counter = new AtomicInteger(0);
+        while ( pointer != null ) {
+            counter.incrementAndGet();
+            CompletableFuture<String> cf = streamManager.fetchEvent(pointer, serializer);
+            cf.thenAccept(s -> {
```

Comment:
You may want to collect these futures so that at the end you can assert that they all complete.